pkg/generate/code/set_resource_test.go

```diff
@@ -3273,6 +3273,336 @@ func TestSetResource_EC2_DHCPOptions_NestedSetConfig(t *testing.T) {
 	)
 }
 
+func TestSetResource_EventBridge_Rule_SetResourceForStruct(t *testing.T) {
+	assert := assert.New(t)
+	require := require.New(t)
+
+	g := testutil.NewModelForService(t, "eventbridge")
+
+	crd := testutil.GetCRDByName(t, g, "Rule")
+	require.NotNil(crd)
+
+	// Getting CRD Target shape
+	field, ok := crd.SpecFields["Targets"]
+	require.True(ok)
+	koShape := &field.ShapeRef.Shape.MemberRef
+
+	// Getting SDK Target shape
+	putTargetsOP, ok := g.SDKAPI.API.Operations["PutTargets"]
+	require.True(ok)
+	targetsShape, ok := putTargetsOP.InputRef.Shape.MemberRefs["Targets"]
+	require.True(ok)
+	targetShape := &targetsShape.Shape.MemberRef
+
+	expected := `	if resp.Arn != nil {
+		ko.ARN = resp.Arn
+	}
+	if resp.BatchParameters != nil {
+		kof1 := &svcapitypes.BatchParameters{}
+		if resp.BatchParameters.ArrayProperties != nil {
+			kof1f0 := &svcapitypes.BatchArrayProperties{}
+			if resp.BatchParameters.ArrayProperties.Size != nil {
+				kof1f0.Size = resp.BatchParameters.ArrayProperties.Size
+			}
+			kof1.ArrayProperties = kof1f0
+		}
+		if resp.BatchParameters.JobDefinition != nil {
+			kof1.JobDefinition = resp.BatchParameters.JobDefinition
+		}
+		if resp.BatchParameters.JobName != nil {
+			kof1.JobName = resp.BatchParameters.JobName
+		}
+		if resp.BatchParameters.RetryStrategy != nil {
+			kof1f3 := &svcapitypes.BatchRetryStrategy{}
+			if resp.BatchParameters.RetryStrategy.Attempts != nil {
+				kof1f3.Attempts = resp.BatchParameters.RetryStrategy.Attempts
+			}
+			kof1.RetryStrategy = kof1f3
+		}
+		ko.BatchParameters = kof1
+	}
+	if resp.DeadLetterConfig != nil {
+		kof2 := &svcapitypes.DeadLetterConfig{}
+		if resp.DeadLetterConfig.Arn != nil {
+			kof2.ARN = resp.DeadLetterConfig.Arn
+		}
+		ko.DeadLetterConfig = kof2
+	}
+	if resp.EcsParameters != nil {
+		kof3 := &svcapitypes.EcsParameters{}
```

Comment:
Hmm, this should be `ECSParameters`... we need to [add ECS to the initialisms](https://github.com/aws-controllers-k8s/pkg/blob/main/names/names.go#L101).